### PR TITLE
Use prompter and handle deletions

### PR DESCRIPTION
**Pull request by johnpc**
#### Description of changes

This PR makes a couple small changes:

1. it uses `amplify-prompter` in areas where we've noticed timeouts caused by unexpected prompts when the `--yes` flag is set
2. it adds some error checking on delete - if the error is a not found exception, it treats the deletion as a success
3. it adds a timeout bump to the artifact scanning script in our e2e pipeline. 10m to scan and clean was not enough and caused flakiness.

#### Issue #, if available

n/a

#### Description of how you validated changes

* ran cloud e2e https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/7468/workflows/d5cb0b96-da72-410e-aaaa-ad2df4d4f27a
* manually tested prompts with `--yes` flag
* manually tested deletions

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts [hunk 1]
```diff
@@ -499,8 +500,8 @@ export const getTriggerEnvInputs = async (context, triggerPath, triggerKey, trig
             (Object.keys(currentEnvVars) && Object.keys(currentEnvVars).length === 0) ||
             !currentEnvVars[questions[j].key]
           ) {
-            const answer: any = await inquirer.prompt(questions[j].question);
-            answers[questions[j].key] = answer[questions[j].key];
+            const answer: any = await prompter[questions[j].question.type](questions[j].question.message);
```
[the remark this answers is not in the document]
Comment (johnpc) [reply packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts:503]: They are all input, except for the interactions category: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-category-interactions/src/provider-utils/supported-services.json#L43

However, all e2e tests pass and I am able to add interactions with no issues, so I believe this function is not used for interactions.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { $TSContext, $TSObject, exitOnNextTick, ResourceCredentialsNotFoundError, ResourceDoesNotExistError } from 'amplify-cli-core';
-import { printer } from 'amplify-prompts';
+import { printer, prompter } from 'amplify-prompts';
 import chalk from 'chalk';
 import { DataApiParams } from 'graphql-relational-schema-transformer';
 import inquirer from 'inquirer';
```
Comment (ammarkarachi) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts:5]: Do you still need this in the file?

**Comment by ammarkarachi** [comment]
LGTM just some unused

**Review by ammarkarachi — APPROVED** [review]

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts (hunk 2 above)
Comment (johnpc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts:5]: yes - it used on line #246 https://github.com/aws-amplify/amplify-cli/pull/10122/files#diff-6e8be270be62c47a5ef68d95acd1d985d857cc77917d2d391517c288a5fc9eafR246

File: packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnpc) [reply packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts]: Updated with mapping to handle alternate types:

```
            const prompterTypeMapping = {
              input: 'input',
              list: 'pick',
              confirm: 'yesOrNo',
            }
            const prompterFunction = prompterTypeMapping[questions[j].question.type];
            const answer: any = await prompter[prompterFunction](questions[j].question.message);
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts (hunk 2 above)
Comment (johnpc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts]: removed :)

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**